impeller/display_list/dl_dispatcher.h

```diff
@@ -353,12 +362,17 @@ class TextFrameDispatcher : public flutter::IgnoreAttributeDispatchHelper,
   // |flutter::DlOpReceiver|
   void setImageFilter(const flutter::DlImageFilter* filter) override;
 
+  std::unordered_map<int64_t, BackdropData> TakeBackdropData() {
+    return std::move(backdrop_keys_);
+  }
```

Comment:
This leaves the instance variable in an invalid state.
```suggestion
  std::unordered_map<int64_t, BackdropData> TakeBackdropData() {
    std::unordered_map<int64_t, BackdropData> temp;
    std::swap(temp, backdrop_keys_);
    return temp;
  }
```

Reply:
Is that really true? seems like a bad API then but I guess that is par for the course for C++. THis is called right before we destroy the TextFrameIterator anyway.

Reply:
I kind of misspoke, it's not that it's in an invalid state, but it's in an undefined state.  Like it is safe to delete it but the contents are not guaranteed to be empty.  https://stackoverflow.com/questions/17730689/is-a-moved-from-vector-always-empty. 

In any case, since we can't guarantee that the unordered_map is not used again we should leave it in a defined state.

Reply:
Damn C++ you scary

Reply:
Done